packages/react-shopify-app-route-propagator/src/hook.ts

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import {ClientApplication} from '@shopify/app-bridge';
+import {History} from '@shopify/app-bridge/actions';
+import {LocationOrHref} from './types';
+
+import {
+  getSelfWindow,
+  getTopWindow,
+  getOrigin,
+  MODAL_IFRAME_NAME,
+} from './globals';
+
+export default function useRoutePropagation(
+  app: ClientApplication<any>,
+  location: LocationOrHref,
+) {
+  const history = React.useRef(History.create(app));
```

Comment:
It's not technically safe to do this because `app` changing break this, but maybe that's OK?